### ISSUE TITLE
Tests #147 and #152 fail in Fedora

### ISSUE THREAD

**Issue by mikelolasagasti** [bats-core/bats-core#664]
Tests `not ok 147 junit formatter: escapes xml special chars` and `not ok 152 junit does not mark tests with FD 3 output as failed (issue #360)` are failing currently during Fedora build.

**To Reproduce**
1. run `./bin/bats test` in Fedora

**Environment (please complete the following information):**
 - Bats Version: 1.8.0
 - Fedora Rawhide & Fedora 36
 - Bash version: 5.2
 - Parallel version: 20220922

**Additional context**

bats-core tests started to fail after bash-5.2 and parallel-20220922 update.

Koschei information: https://koschei.fedoraproject.org/package/bats?collection=f38

Fedora rawhide build log: https://kojipkgs.fedoraproject.org/work/tasks/4987/92774987/build.log

```
not ok 147 junit formatter: escapes xml special chars
# (in test file test/junit-formatter.bats, line 56)
#   `[[ "${lines[2]}" == "<testsuite name=\"$ESCAPED_TEST_FILE_NAME\" tests=\"3\" failures=\"1\" errors=\"0\" skipped=\"1\" time=\""*"\" timestamp=\""*"\" hostname=\""*"\">" ]]' failed
# <?xml version="1.0" encoding="UTF-8"?>
# <testsuites time="0.017">
# <testsuite name="xml-escape-"quot;<lt;>gt;'#39;&amp;.bats" tests="3" failures="1" errors="0" skipped="1" time="0.017" timestamp="2022-10-08T01:47:12" hostname="ade0001bea794ee09b9c346605ac4954">
#     <testcase classname="xml-escape-"quot;<lt;>gt;'#39;&amp;.bats" name="Successful test with escape characters: "quot;'#39;<lt;>gt;&amp;#27;[0m (0x1b)" time="0.009" />
#     <testcase classname="xml-escape-"quot;<lt;>gt;'#39;&amp;.bats" name="Failed test with escape characters: "quot;'#39;<lt;>gt;&amp;#27;[0m (0x1b)" time="0.008">
#         <failure type="failure">(in test file /tmp/bats-run-pDKmcA/test/147/xml-escape-"quot;<lt;>gt;'#39;&amp;.bats, line 6)
#   `echo "quot;<lt;>gt;'#39;&amp;#27;[0m"quot; &amp;&amp; false'#39; failed
# <lt;>gt;'#39;&amp;#27;[0m</failure>
#     </testcase>
#     <testcase classname="xml-escape-"quot;<lt;>gt;'#39;&amp;.bats" name="Skipped test with escape characters: "quot;'#39;<lt;>gt;&amp;#27;[0m (0x1b)" time="0">
#         <skipped>"quot;'#39;<lt;>gt;&amp;#27;[0m</skipped>
#     </testcase>
#
# </testsuite>
# </testsuites>
``` 

```
not ok 152 junit does not mark tests with FD 3 output as failed (issue #360)
# (in test file test/junit-formatter.bats, line 133)
#   `[[ "${lines[13]}" == '  `false&#39; failed' ]]' failed
# <?xml version="1.0" encoding="UTF-8"?>
# <testsuites time="0.023">
# <testsuite name="issue_360.bats" tests="2" failures="1" errors="0" skipped="0" time="0.023" timestamp="2022-10-08T01:47:13" hostname="ade0001bea794ee09b9c346605ac4954">
#     <testcase classname="issue_360.bats" name="say hello to Biblo" time="0.011">
#         <system-out>setup FD3
# hello Bilbo
# teardown FD3</system-out>
#     </testcase>
#     <testcase classname="issue_360.bats" name="fail to say hello to Biblo" time="0.012">
#         <system-out>setup FD3
# hello Bilbo
# teardown FD3</system-out>
#         <failure type="failure">(in test file test/fixtures/junit-formatter/issue_360.bats, line 21)
#   `false'#39; failed
# # setup stdout
# # hello stdout
# # teardown stdout</failure>
#     </testcase>
#
# </testsuite>
# </testsuites>
``` 

Fedora bats-core spec: https://src.fedoraproject.org/rpms/bats/blob/rawhide/f/bats.spec

**Comment by martin-schulze-vireso** [comment]
Should be fixed with #656 can you try with current master?

**Comment by mikelolasagasti** [comment]
Thanks for the fast reply @martin-schulze-vireso I checked latest commits but somehow missed that one, sorry for the noise.

New builds have worked after applying the patch.
